crd/apis/network/v1/network.go

```diff
@@ -26,3 +28,21 @@ func (n *Network) InUse() bool {
 	val, ok := n.Annotations[NetworkInUseAnnotationKey]
 	return ok && val == NetworkInUseAnnotationValTrue
 }
+
+// GetCondition returns the "condType" condition or nil if not found.
+func (n *Network) GetCondition(condType NetworkConditionType) *metav1.Condition {
```

Comment:
why do you add methods to the object instead creating functions that receive the object?
it is more scalable and more portable, since you don't depend on the object version

Reply:
I find this more convenient, but I agree that conditions are cross objects and we could reuse same functions on the other ones as well. I have changed the code to address this.